lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1097,12 +930,19 @@ def reset_stat(tarinfo):
                         continue
 
                 # Dealing with a normal file, just add it by name.
-                tar_file.add(os.path.realpath(b_src_path), arcname=filename, recursive=False, filter=reset_stat)
+                tar_file.add(
+                    to_native(os.path.realpath(b_src_path)),
```

Comment:
is there a reason why you have a native str, I would have assumed it was a byte str beforehand for a reason. I know the tarfile is very particular about the type of string being used.

Reply:
I don't remember why this was necessary. I seem to recall it's unicode under py3...

Reply:
Oh, wait! I think I remember. It was mypy that hinted that the type is not right and so I checked this in docs / stubs / source / https://github.com/python/typeshed/blob/869238e/stdlib/2and3/tarfile.pyi#L162-L189 / https://github.com/python/cpython/blob/master/Lib/tarfile.py and this is what was necessary.